src/libcommonserver/io/iohelper_win.cpp

```diff
@@ -329,126 +329,251 @@ bool IoHelper::checkIfFileIsDehydrated(const SyncPath &itemPath, bool &isDehydra
     return IoHelper::getXAttrValue(itemPath.native(), FILE_ATTRIBUTE_OFFLINE, isDehydrated, ioError);
 }
 
+static bool setRightsWindowsApi(const SyncPath &path, DWORD permission, ACCESS_MODE accessMode, IoError &ioError,
+                                log4cplus::Logger logger) noexcept {  // Always return false if ioError is not IoErrorSuccess,
+                                                                      // caller should check _isExpectedError(ioError)
+    PACL pACL_old = nullptr;  // Current ACL
+    PACL pACL_new = nullptr;  // New ACL
+    PSECURITY_DESCRIPTOR pSecurityDescriptor = nullptr;
+    EXPLICIT_ACCESS ExplicitAccess;
+    ZeroMemory(&ExplicitAccess, sizeof(ExplicitAccess));
+
+    ExplicitAccess.grfAccessPermissions = permission;
+    ExplicitAccess.grfAccessMode = accessMode;
+    ExplicitAccess.grfInheritance = NO_INHERITANCE;
+    ExplicitAccess.Trustee.pMultipleTrustee = nullptr;
+    ExplicitAccess.Trustee.MultipleTrusteeOperation = NO_MULTIPLE_TRUSTEE;
+    ExplicitAccess.Trustee.TrusteeForm = Utility::_trustee.TrusteeForm;
+    ExplicitAccess.Trustee.TrusteeType = Utility::_trustee.TrusteeType;
+    ExplicitAccess.Trustee.ptstrName = Utility::_trustee.ptstrName;
+
+    LPCWSTR pathw_c = Path2WStr(path).c_str();
```

Comment:
Use a variable to store Path2WStr(path)

Reply:
Done in commit: https://github.com/Infomaniak/desktop-kDrive/pull/68/commits/6472cc919065de890487e746a983781ea479f403